atlas_core/data_import.py

```diff
@@ -87,30 +41,76 @@ def import_data(file_name="./data.h5", engine=None, source_chunksize=10**6,
                 df = classification_to_pandas(df)
-                df.to_sql(table_name, engine, index=False,
-                          chunksize=dest_chunksize, if_exists="append")
+                df.to_sql(
+                    table_name,
+                    engine,
+                    index=False,
+                    chunksize=dest_chunksize,
+                    if_exists="append",
+                )
 
             else:
-
                 # If it's a timeseries data table, load it in chunks to not
                 # exhaust memory all at once
-                iterator = pd.read_hdf(file_name, key=key,
-                                       chunksize=source_chunksize,
-                                       iterator=True)
+                iterator = pd.read_hdf(
+                    file_name, key=key, chunksize=source_chunksize, iterator=True
+                )
 
                 for i, df in enumerate(iterator):
                     print(i * source_chunksize)
 
                     # Add in level fields
                     if "levels" in metadata:
                         for entity, level_value in metadata["levels"].items():
-                            df[entity+"_level"] = level_value
+                            df[entity + "_level"] = level_value
 
-                    df.to_sql(table_name, engine, index=False,
-                              chunksize=dest_chunksize, if_exists="append")
+                    df.to_sql(
+                        table_name,
+                        engine,
+                        index=False,
+                        chunksize=dest_chunksize,
+                        if_exists="append",
+                    )
 
                     # Hint that this object should be garbage collected
                     del df
 
         except SQLAlchemyError as exc:
             print(exc)
+
+
+def import_data(
+    file_name="./data.h5",
+    engine=None,
+    source_chunksize=10 ** 6,
+    dest_chunksize=10 ** 6,
+    keys=None,
+    database="postgres",
+):
+    """Import data from a data.h5 (i.e. HDF) file into the SQL DB. This
+    needs to be run from within the flask app context in order to be able to
+    access the db engine currently in use.
+
+    In the HDF store, we expect to find one table per SQL table, with specific
+    attributes:
+
+        - sql_table_name: tells us which table to write to. If it doesn't
+        exist, we skip reading this table.
+        - product_level: modifies the "level" column, for cases when multiple
+        levels of data (e.g. 2digit and 4digit) get loaded into a single table
+
+    In addition, anything under the /classifications/ path in the HDF store
+    gets treated specially as a classification, and gets run through the
+    classification_to_pandas() function.
+    """
+
+    if database == "postgres":
+        from hdf_to_postgres import hdf_to_postgres
+
+        # No engine defined as a kwarg for postgres
```

Comment:
Worth specifying why - the main reason is because we instead pass in engine_args and engine_kwargs because the workers have to construct their own engines. Would be nice to expose and pass in engine_args and engine_kwargs at this layer too - use either the former or the latter, and perhaps specify in the docs.